### PR TITLE
Create candidate bio page

### LINKED ISSUE #48: 🤵‍♂️ Candidate Bio
**Type**: client
**FOF**:  2

## Description
- Build out the page that overviews a candidate for a specific race

## Screenshot
image

## Todo

- [ ] Create candidate bio page using dummy data
- [ ] Integrate button that allows user to favorite candidate

## Links
- [Figma](https://www.figma.com/file/6aJ3cfu2Yq5dxUQcNDkm6Y/Brink-Designs?node-id=34%3A2)

### PR DESCRIPTION
**Pull request by ericcccsliu**
## Status:

:construction: In development

## Summary

Closes #48 

## Changes

- Move favorite star to separate component 
- Create candidate bio page 

## Screenshots
Screen Shot 2022-03-18 at 5 48 35 PM

### CODE REVIEW

File: client/src/components/button/FavoriteButton.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import { Button, Icon } from '@chakra-ui/react';
+import { Star, StarBorder } from '@mui/icons-material';
+import React, { FC, ReactElement, useState } from 'react';
+
+import { useFavoritedCandidates } from '../../contexts/FavCandidatesContext';
```
Comment (amit-sawhney) [inline client/src/components/button/FavoriteButton.tsx:5]: did we not add this as an export that you could get from just `../contexts`

File: client/src/components/button/FavoriteButton.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+interface FavoriteCandidateProps {
+  name: string;
+  defaultChecked?: boolean;
+  [key: string]: any;
```
Comment (amit-sawhney) [inline client/src/components/button/FavoriteButton.tsx:10]: we fixed this on Sue's PR, but this is bad practice don't do.

File: client/src/components/card/CandidateCard.tsx [hunk 3]
```diff
@@ -1,10 +1,9 @@
...
 import type { BoxProps } from '@chakra-ui/react';
-import { Star, StarBorder } from '@mui/icons-material';
-import React, { FC, ReactElement, useState } from 'react';
+import React, { FC, ReactElement } from 'react';
 
-import { useFavoritedCandidates } from '../../contexts/FavCandidatesContext';
+import FavoriteButton from '../button/FavoriteButton';
 
 interface CandidateCardProps extends BoxProps {
   name: string;
```
Comment (amit-sawhney) [inline client/src/components/card/CandidateCard.tsx:9]: Can we generalize this to use the Candidate type now that we have some of those types

File: client/src/components/card/CandidateCard.tsx [hunk 4]
```diff
@@ -16,65 +15,34 @@ export const CandidateCard: FC<CandidateCardProps> = ({
   name,
   party,
   defaultChecked,
-  ...rest
```
Comment (amit-sawhney) [inline client/src/components/card/CandidateCard.tsx:19]: Did you mean to take out rest and not use it anymore?

File: client/src/components/link/SocialLink.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
+import React, { FC, ReactElement } from 'react';
+import TwitterIcon from '@mui/icons-material/Twitter';
+import FacebookIcon from '@mui/icons-material/Facebook';
+import LinkIcon from '@mui/icons-material/Link';
+import { Flex, Icon, Link, Heading } from '@chakra-ui/react';
+import type { LinkProps } from '@chakra-ui/react';
+
+import type { CandidateUrl } from '../../types/ballotReady';
+
+interface UrlType extends LinkProps {
```
Comment (amit-sawhney) [inline client/src/components/link/SocialLink.tsx:10]: Rename to SocialLinkProps

File: client/src/components/link/SocialLink.tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+      direction="row"
+      justify="flex-start"
+      align="center"
+      width="165px"
+      height="55px"
+      bg="lightGrey"
+      borderRadius="5px"
+    >
+      <Icon
+        as={urlPairs[candidateUrl.type].icon}
```
Comment (amit-sawhney) [inline client/src/components/link/SocialLink.tsx:44]: very sexy

File: client/src/layout/Navbar.scss [hunk 7]
```diff
@@ -0,0 +1,55 @@
+@import '../styles/styles.scss';
```
Comment (amit-sawhney) [inline client/src/layout/Navbar.scss:1]: wait where did this file come from -- i thought we scrapped most of this? Like why is it an entirely new file

File: client/src/pages/CandidateBio.tsx [hunk 8]
```diff
@@ -0,0 +1,90 @@
+import React, { FC, ReactElement } from 'react';
+import { Box, Flex, Text, Heading, Image } from '@chakra-ui/react';
+import type { FlexProps } from '@chakra-ui/react';
+// import TwitterIcon from '@mui/icons-material/Twitter';
+// import FacebookIcon from '@mui/icons-material/Facebook';
+// import LinkIcon from '@mui/icons-material/Link';
```
Comment (amit-sawhney) [inline client/src/pages/CandidateBio.tsx:6]: Delete unused comments

File: client/src/pages/CandidateBio.tsx [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+
+import type { Candidate } from '../types/ballotReady';
+import FavoriteButton from '../components/button/FavoriteButton';
+import { SocialLink } from '../components/link/SocialLink';
```
Comment (amit-sawhney) [inline client/src/pages/CandidateBio.tsx:10]: These should be imported from `../components`

File: client/src/utils/constants.ts [hunk 10]
```diff
@@ -15,4 +15,5 @@ export const APP_ROUTE = {
   CHECKLIST: 'checklist',
   FAQ: 'faq',
   BALLOT: 'ballot',
+  BIO: 'bio',
```
Comment (amit-sawhney) [inline client/src/utils/constants.ts:18]: Yea make sure to delete all of the routes that we did this way - I don't think it's cleaner tbh

File: yarn.lock [hunk 11]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (amit-sawhney) [inline yarn.lock:1]: Uh did you push a yarn.lock to the root?

**Review by amit-sawhney — CHANGES_REQUESTED** [review]
Just some questions to address first, but looking good :)

File: client/src/components/card/CandidateCard.tsx (hunk 4 above)
Comment (ericcccsliu) [reply client/src/components/card/CandidateCard.tsx:19]: oops just put it back in

File: yarn.lock (hunk 11 above)
Comment (ericcccsliu) [reply yarn.lock:1]: oops maybe

**Review by amit-sawhney — APPROVED** [review]

**Pull request merged**